include/roboteam_ai/skills/CoachDefend.h

```diff
@@ -15,10 +16,13 @@ namespace ai {
 class CoachDefend : public Skill {
     private:
         bool useBasicGtp(Vector2 targetLocation);
+        rtt::ai::coach::DefenceDealer defenceDealer;
```

Comment:
This is not how to do it. You're just creating a local coaches for each robot now. The coaches coordinate global behavior so this is not going to work.

Reply:
Oops. You're right. I thought coaches acted as a sort of polling structure, containing mostly information on where the robots should go if they didn't know what to do, and telling them if they're being passed to. However, the coaches are much more than that, and are changed a lot when they are used. Realizing this, I'm not sure how we can improve this design whilst keeping them alive.